### PR TITLE
Vehicle spawner

### PR DESCRIPTION
**Pull request by JoramD**
**When merged this pull request will:**
- Add vehicle spawner
- Add entry in `featureOverview`

**To be merged after next modpack update.**

### CODE REVIEW

File: Olympus.Stratis/functions/fn_featureOverview.sqf [hunk 1]
```diff
@@ -20,6 +20,11 @@ _player createDiaryRecord ["features", ["Environment Controller", "
     <br/>The main base and some courses have an additional laptop to control the environment.
 "]];
 
+_player createDiaryRecord ["features", ["Vehicle Spawner", "
+    <br/><img image='x\tacr\addons\images\olympus\feature\vehicleSpawn.jpg' width='300' height='350'/>
```
Comment (JoramD) [inline Olympus.Stratis/functions/fn_featureOverview.sqf:24]: Currently has an issue with this image, it won't even work on a laptop (while the teleporter one from #127 works fine.

File: Olympus.Stratis/functions/fn_vehicleSpawner.sqf [hunk 2]
```diff
@@ -56,6 +56,9 @@ private _spawnAction = [
                     if (count (_spawnPos nearEntities 5) == 1) then {
                         private _spawnedVehicle = createVehicle [_classname, _spawnPos, [], 0, "CAN_COLLIDE"];
                         _spawnedVehicle setDir 60;
+                        _spawnedVehicle adDItemCargoGlobal ["ToolKit", 1];
```
Comment (jonpas) [inline Olympus.Stratis/functions/fn_vehicleSpawner.sqf]:
```suggestion
                        _spawnedVehicle addItemCargoGlobal ["ToolKit", 1];
```

File: Olympus.Stratis/functions/fn_vehicleSpawner.sqf [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            _x params ["_vehicle", "_playerName", "_vehicleFancyName"];
+
+            private _removeAction = [
+                format [QGVAR(removeAction_%1), _vehicle],
+                format ["Remove %1 (%2)", _vehicleFancyName, _playerName],
+                "",
+                {
+                    (_this select 2) params ["_controller", "_vehicle", "_playerName", "_vehicleFancyName"];
+
+                    if (count (crew _vehicle) == 0) then {
```
Comment (jonpas) [inline Olympus.Stratis/functions/fn_vehicleSpawner.sqf]:
```suggestion
                    if ((crew _vehicle) isEqualTo []) then {
```

File: Olympus.Stratis/functions/fn_vehicleSpawner.sqf [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            ["tacs_Arcadian_B_Black", "Arcadian GT (Black)"],
+            ["tacs_Arcadian_B_Green", "Arcadian GT (Green)"],
+            ["tacs_Arcadian_B_Tan", "Arcadian GT (Tan)"],
+            ["tacs_BWA3_B_Eagle_Black", "Eagle IV (Black)"],
+            ["B_LSV_01_unarmed_F", "Polaris DAGOR"],
+            ["C_SUV_01_F", "SUV"],
+            ["C_Offroad_01_F", "Offroad"],
+            ["C_Offroad_01_covered_F", "Offroad (Covered)"],
+            ["C_Offroad_02_unarmed_F", "Jeep Wrangler"],
+            ["CUP_C_Golf4_kitty_Civ", "Golf"], // Tyrone made me do this
```
Comment (jonpas) [inline Olympus.Stratis/functions/fn_vehicleSpawner.sqf]: Remove. :dagger:

File: Olympus.Stratis/functions/fn_vehicleSpawner.sqf [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    "Spawn Vehicle",
+    "",
+    {},
+    {true},
+    {
+        (_this select 2) params ["_controller", "_spawnPos"];
+
+        private _allowedVehicles = [
+            // ["classname", "fancyName"]
+            ["tacs_Arcadian_B_Black", "Arcadian GT (Black)"],
```
Comment (jonpas) [inline Olympus.Stratis/functions/fn_vehicleSpawner.sqf]: Why do you need a fancy name, read it from the config by classname.

File: Olympus.Stratis/functions/fn_vehicleSpawner.sqf [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+ * None
+ *
+ * Example:
+ * [this, spawnPad] call TAC_Olympus_fnc_vehicleSpawner
+ */
+
+params ["_controller", "_spawnPos"];
+
+// Make empty marker for featureOverview
+createMarker ["vehicleSpawnerMarker", _controller];
```
Comment (jonpas) [inline Olympus.Stratis/functions/fn_vehicleSpawner.sqf]: Needs to be prefixed.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: Olympus.Stratis/functions/fn_featureOverview.sqf (hunk 1 above)
Comment (JoramD) [reply Olympus.Stratis/functions/fn_featureOverview.sqf:24]: Solved, it was a corrupted image (please never use paint again @TyroneMF)

**Review by jonpas — APPROVED** [review]

**Review by Mike-MF — APPROVED** [review]

**Pull request merged**